components/event-publisher-proxy/pkg/handler/beb/handler_test.go

```diff
@@ -487,18 +487,16 @@ func validateApplicationName(appName string) testingutils.Validator {
 			if strings.Contains(event.Type(), appName) {
 				return nil
 			}
-			return errors.New(fmt.Sprintf("expected the event-type [%s] to contain the application name [%s] for structured mode", event.Type(), appName))
+			return fmt.Errorf("expected the event-type [%s] to contain the application name [%s] for structured mode", event.Type(), appName)
 		}
-		// default CE binary mode
-		for k, v := range r.Header {
-			if strings.ToLower(k) != strings.ToLower(testingutils.CeTypeHeader) {
-				continue
-			}
-			if strings.Contains(v[0], appName) {
-				return nil
-			}
-			return errors.New(fmt.Sprintf("expected the event-type [%s] to contain the application name [%s] for binary mode", v[0], appName))
+		// Default CE binary mode
+		value, ok := r.Header[testingutils.CeTypeHeader]
```

Comment:
IMO r.Header.Get() fits well here as it takes of the case as well.

Reply:
Thought about the same but not sure about empty headers vs non-existing headers. `Get()` would return an empty string. But might be okay here for the test, so I'll change.